python_modules/dagster/dagster/_core/definitions/time_window_partitions.py

```diff
@@ -106,20 +130,27 @@ def __new__(
         cls,
         start: Union[datetime, str],
         fmt: str,
-        end: Union[datetime, str, None] = None,
-        schedule_type: Optional[ScheduleType] = None,
         timezone: Optional[str] = None,
+        end: Union[datetime, str, None] = None,
         end_offset: int = 0,
+        cron_schedule: Optional[str] = None,
+        # Fields below will not be serialized
+        schedule_type: Optional[ScheduleType] = None,
         minute_offset: Optional[int] = None,
         hour_offset: Optional[int] = None,
         day_offset: Optional[int] = None,
-        cron_schedule: Optional[str] = None,
     ):
         check.opt_str_param(timezone, "timezone")
         timezone = timezone or "UTC"
 
         if isinstance(start, datetime):
             start_dt = pendulum.instance(start, tz=timezone)
+
+            if start.tzinfo:
```

Comment:
pendulum.instance does not convert the provided datetime to the correct timezone, we need to handle this manually.

This will make a `2023-01-01T05:00:00+00:00` into a `2023-01-01T00:00:00-05:00`